aws/src/index.js

```diff
@@ -819,35 +822,67 @@ const handleRecords = (event, context, lintEnv) => {
 };
 
 const handler = (event, context) => {
-  const apiToken = process.env.LogIntelligence_API_Token;
-  if (!apiToken) {
-    handleError('The API token is missing. Please configure it in an environment variable of the lambda function');
-    return;
-  }
-
-  const ingestionUrl = process.env.LogIntelligence_API_Url || 'https://data.mgmt.cloud.vmware.com/le-mans/v1/streams/ingestion-pipeline-stream';
-
-  const tagRegexMap = new Map();
-  Object.getOwnPropertyNames(process.env).forEach((v) => {
-    if (v.startsWith('Tag_')) {
-      tagRegexMap.set(v.substring(4), new RegExp(process.env[v], 'i'));
-    }
-  });
-
-  const lintEnv = new LIntHttpEnv(`Bearer ${apiToken}`, ingestionUrl);
-
-  if (event.awslogs) {
-    handleCloudWatchLogs(event, context, lintEnv, tagRegexMap);
-  }
+    let apiToken = null;
+    let lintEnv = null;
+    const ingestionUrl = process.env.VRLIC_API_Url || 'https://data.mgmt.cloud.vmware.com/le-mans/v1/streams/ingestion-pipeline-stream';
+
+    const tagRegexMap = new Map();
+    Object.getOwnPropertyNames(process.env).forEach((v) => {
+        if (v.startsWith('Tag_')) {
+            tagRegexMap.set(v.substring(4), new RegExp(process.env[v], 'i'));
+        }
+    });
+    if (process.env.VAULT_ADDR || process.env.VLE_VAULT_ADDR) {
+        const options = {
+            hostname: '127.0.0.1',
+            port: 8200,
+            path: '/v1/' + process.env.KV_SECRET_PATH,
+            method: 'GET'
+        }
+        readVaultData(options).then(function(token) {
+            apiToken = token;
+            if (!apiToken) {
+                handleError('The API token is missing in vault.');
+                return;
+            }
+            lintEnv = new LIntHttpEnv(`Bearer ${apiToken}`, ingestionUrl);
+
+            if (event.awslogs) {
+                handleCloudWatchLogs(event, context, lintEnv, tagRegexMap);
+            }
+
+            if (event.Records || event.records) {
+                handleRecords(event, context, lintEnv);
+            }
+
+            if (process.env.EventBridge_Logs === 'true') {
+                handleEventBridgelogs(event, context, lintEnv);
+            }
+        }).catch(function(err) {
+            console.log(err);
+        });
+    } else if (process.env.VRLIC_API_Token) {
+        apiToken = process.env.VRLIC_API_Token;
+        if (!apiToken) {
+            handleError('The API token is missing. Please configure it in an environment variable of the lambda function');
+            return;
+        }
+        lintEnv = new LIntHttpEnv(`Bearer ${apiToken}`, ingestionUrl);
 
-  if (event.Records || event.records) {
-    handleRecords(event, context, lintEnv);
-  }
+        if (event.awslogs) {
```

Comment:
why is this change done now? did we see any issues in the previous code?

Reply:
just provided the condition on behalf of vault integration, Didn't modify the previous code.